### ISSUE TITLE
read.adp.rdi(): recognize code 0x00 0x30 "Variable Fixed Attitude"

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1859]
I've split this task out from #1858 because otherwise that issue will become a mess of notes on the sub-tasks.

The goal here is to handle code 0x00 0x30, i.e. `Binary Fixed Attitude Data`, as described in
* table 35, page 150 of ref 1
* figure 52 (overview) and table 45 (details), pages 172-173 of ref 1

**References.**

Ocean Surveyor Technical Manual_Dec20.pdf, P/N 95A-6012-00 (December 2020)

**Comment by dankelley** [comment]
Below is a screenshot of the relevant part, so I and other developers (and users) won't have to find the information in the RDI documentation.

<img width="797" alt="Screen Shot 2021-08-06 at 3 06 45 PM" src="https://user-images.githubusercontent.com/99469/128553627-a005d676-5454-474c-a444-29aaa3779018.png">

**Comment by dankelley** [comment]
The entries in this table seem like `metadata` to me, not like `data`, in the sense that I don't think they will vary from one ensemble to the next.  (I'll check that.)

**Comment by dankelley** [comment]
This is *very* detailed stuff.  We do not support actually *using* any of this information (e.g. in perhaps adjusting the measured fields in some ways), but I'll have a go at reading it, and storing it in the `metadata` slot.  That way, if there is ever a user who wants to look, they can.

**Comment by dankelley** [comment]
I instrumented the code to detect the 0x00 0x30 sequence, and then made it print out the 34 bytes of contents (according to my understanding of Table 45 of reference 1), and I get as below.  So it looks like we *have* this header, but that all the data contained in it are 0, for this particular file.  That will make it a bit hard to do reality checks on the contents (which I'd like so as to get some idea of whether I'm splitting the bytes in the right places) but I suppose other files can be checked, for that.

```
        Reading 'Binary Fixed Attitude Header' in first 3 ensembles...
        ensemble:  1 
          ensembleStart[ 1 ]= 1 
          header$dataOffset[ 10 ]= 2433 
          therefore J= 2434  (sum of previous two numbers)
          buf[J+1:34]:  30 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
        ensemble:  2 
          ensembleStart[ 2 ]= 2580 
          header$dataOffset[ 10 ]= 2433 
          therefore J= 5013  (sum of previous two numbers)
          buf[J+1:34]:  30 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
        ensemble:  3 
          ensembleStart[ 3 ]= 5159 
          header$dataOffset[ 10 ]= 2433 
          therefore J= 7592  (sum of previous two numbers)
          buf[J+1:34]:  30 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
```

**Comment by dankelley** [comment]
I cannot find anything in the docs to indicate how to read these things.  For example, bytes 3-10 are said to hold "user input for the Variable Attitude data to be output".  Well, that's 8 bytes.  So maybe it's an 8-byte floating point value?  If so, in what endian?  Or maybe it's a long-int (again, in what endian?).  With a case where all items are 0, it's hard to do sensibility tests (e.g. on signs of things) so this becomes a guessing game on
* which endian form is used?
* is the thing int or float?
* is the thing signed or not?

And it's worse, later on.  The thing called "Fixed Heading Scaling" is in bytes 12-14, so that is **three** bytes. What???  I've never heard of a 3-byte int or float, on any modern machine.  (Nor on any ancient machine, to be honest.)

My inclination is just to store in the metadata the raw bytes of this item.  That way, if we get a user who knows what these things mean, they can let us know and we can code in something that's not just a crazy guess.

I will do this tomorrow before noon.  I don't see much loss in this decision, frankly.  It's better than leaving this issue open for months, just in case some day one of us is reading the manual for fun (!) and notices what we ought to be doing.

Any thoughts on my plan, @richardsc?

PS. the code that produced the output shown in the previous comment is not pushed to GH.

**Comment by dankelley** [comment]
I've addressed this in commit fb321166e90f014c670de6c1a6b64b9b99c3ce0e  of the "develop" branch.  

A test code is at https://github.com/dankelley/oce-issues/blob/main/18xx/1858/1858a.R and my tests with a (confidential) RDI 'Surveyor' file show that (a) this header is, indeed, "fixed" (constant across ensembles, at least for the first 10 that I tested) and (b) the header consists just of the 0x00 0x30 and then a list of 0x00 values.  This means that the test file that I have available is not going to be helpful in providing clues as to how to decode the bytes (and I need advice because the docs are unclear). 

What I am doing now -- and this is explained in a warning -- is to store the raw bytes, in a variable with suitable name.  This means that experts can get access to the bytes, and that if one of those experts wants to help us by explaining the data format, then `read.adp.rdi()` might have a chance of doing the decoding.

Since I'm about to close this issue, as per my plan stated yesterday, I'll show what I get with https://github.com/dankelley/oce-issues/blob/main/18xx/1858/1858a.R using `debug=1` in the read.  (I am redacting the filename.)

```
(redacted filename).ENS
  [["binaryFixedAttitudeHeaderRaw"]]: 30000000000000000000000000000000000000000000000000000000000000000000
Warning messages:
1: In read.adp.rdi(f, debug = 0) :
  provisionally, metadata$binaryFixedAttitudeHeaderRaw holds the 34 raw bytes of the Binary Fixed Attitude header
```
